### DISCUSSION TITLE
pdf using pdfhtml is not searchable

### DISCUSSION THREAD

**Discussion by actsasgeek** [executablebooks/meta#635] [General]
Actually, it's a bit more weird than that!

I generated a book using `--builder pdfhtml` and everything looks just fine (except for the previously mentioned page number issue). However, when I search for something, it isn't found even though I'm looking right at the word. I tested a bit more and found that sometimes terms were found...but always in quotations.

Mystified, I checked to see if I could even copy/paste the text in the PDF. I could but when I did, **it's all backwards**:

.taht yltcaxe uoy hcaet ot ,sraey wef tsal eht revo depoleved ,tpmetta na era seton esruoc esehT

And so I tried searching for "peeD" and it does indeed find "Deep". Looking at the search results is even more fun..
![Screen Shot 2022-01-18 at 1 46 18 PM](https://user-images.githubusercontent.com/216816/149999998-6eef2462-faeb-47e9-91b3-80f7c4820ab1.png)
.

**Comment by actsasgeek** [comment]
Looking into this a bit more, I found this bug report from 2020:

https://github.com/puppeteer/puppeteer/issues/2206

The issue seems to resolve if you use a ttf instead of woff but I'm not sure how this is specified in Jupyter Books. Can we set the kind of font that's used?

**Reply by choldgraf** [reply]
What a weird bug!

I wonder if we should try recommending the use of [playwright's python library](https://playwright.dev/python/docs/intro) instead of pyppetteer. It seems that this is a bit more well-supported and stable 🤔

**Reply by actsasgeek** [reply]
Is it easy to swap in playwright? What about the ttf option?

**Reply by choldgraf** [reply]
It shouldn't be too tough, in my experience Playwright is actually a bit easier to install and to use, but it'd require some updating under the hood.

Here is where we have the `pyppetteer` logic to convert HTML to PDF: https://github.com/executablebooks/jupyter-book/blob/master/jupyter_book/pdf.py#L34

I think we could substitute that logic with a similar approach with Playwright: https://playwright.dev/python/docs/api/class-page#page-pdf

**Reply by actsasgeek** [reply]
so I did a quick test, switching out Playwright for Puppeteer, and...the text is still backwards 😮 

Seems like whatever "it" is, it affects both libraries. Any idea how to switch from `woff` to `ttf`?

**Reply by actsasgeek** [reply]
I'm narrowing in on it. This font information seems to be injected into the index.html that jupyter-book (jb) builds...but I can't figure out from where...

```
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">
```
I tried changing manually and it is definitely regenerated. I looked at the jupyter(lab|notebook) repositories and couldn't quite figure it out. Maybe someone else has an idea?

**Reply by choldgraf** [reply]
I was playing around with getting playwright support in this PR: https://github.com/executablebooks/jupyter-book/pull/1664

Want to try that PR as well as updating to the latest sphinx-book-theme? That also might change how the font files are linked a little bit
